Tools/cases_generator/analyzer.py

```diff
@@ -0,0 +1,470 @@
+    False,
+    True,
+    False,
+    False,
+    False,
+    False,
+    False,
+    False,
+    False,
+)
+
+
+@dataclass
+class Skip:
+    "Unused cache entry"
+    size: int
+
+    @property
+    def name(self) -> str:
+        return f"unused/{self.size}"
+
+    @property
+    def properties(self) -> Properties:
+        return SKIP_PROPERTIES
+
+
+@dataclass
+class StackItem:
+    name: str
+    type: str | None
+    condition: str | None
+    size: str
+    peek: bool = False
+
+    def __str__(self) -> str:
+        cond = " if (" + self.condition + ")" if self.condition else ""
+        size = "[" + self.size + "]" if self.size != "1" else ""
+        type = "" if self.type is None else self.type + " "
+        return f"{type}{self.name}{size}{cond} {self.peek}"
+
+    def is_array(self) -> bool:
+        return self.type == "PyObject **"
+
+
+@dataclass
+class StackEffect:
+    inputs: list[StackItem]
+    outputs: list[StackItem]
+
+    def __str__(self) -> str:
+        return f"({', '.join([str(i) for i in self.inputs])} -- {', '.join([str(i) for i in self.outputs])})"
+
+
+@dataclass
+class CacheEntry:
+    name: str
+    size: int
+
+    def __str__(self) -> str:
+        return f"{self.name}/{self.size}"
+
+
+@dataclass
+class Uop:
+    name: str
+    context: parser.Context | None
+    annotations: list[str]
+    stack: StackEffect
+    caches: list[CacheEntry]
+    body: list[lexer.Token]
+    properties: Properties
+    _size: int = -1
+
+    def dump(self, indent: str) -> None:
+        print(
+            indent, self.name, ", ".join(self.annotations) if self.annotations else ""
+        )
+        print(indent, self.stack, ", ".join([str(c) for c in self.caches]))
+        self.properties.dump("    " + indent)
+
+    @property
+    def size(self) -> int:
+        if self._size < 0:
+            self._size = sum(c.size for c in self.caches)
+        return self._size
+
+
+Part = Uop | Skip
+
+
+@dataclass
+class Instruction:
+    name: str
+    uops: list[Part]
+    _properties: Properties | None
+    is_target: bool = False
+    family: Optional["Family"] = None
+
+    @property
+    def properties(self) -> Properties:
+        if self._properties is None:
+            self._properties = self._compute_properties()
+        return self._properties
+
+    def _compute_properties(self) -> Properties:
+        return Properties(
+            any(part.properties.escapes for part in self.uops),
+            all(part.properties.infallible for part in self.uops),
+            any(part.properties.deopts for part in self.uops),
+            any(part.properties.oparg for part in self.uops),
+            any(part.properties.jumps for part in self.uops),
+            any(part.properties.ends_with_eval_breaker for part in self.uops),
+            any(part.properties.needs_this for part in self.uops),
+            any(part.properties.always_exits for part in self.uops),
+            any(part.properties.stores_sp for part in self.uops),
```

Comment:
The knowledge of which fields exist in properties belong in the Properties class rather than here. You could define a function there that takes a list of Property objects and does this.